packages/p2p-media-loader-hlsjs/src/services/segment-mananger.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  Parser,
+  type PlaylistManifest,
+  type MasterManifest,
+  type Segment as ParserSegment,
+} from "m3u8-parser";
+
+export class SegmentManager {
+  videoPlaylists: VideoPlaylistsContainer = new VideoPlaylistsContainer();
+  audioPlaylists: AudioPlaylistsContainer = new AudioPlaylistsContainer();
```

Comment:
Consider refactoring (simplify) to a map that contains both video and audio playlists: `playlists: Map<string, Playlist>` without abstract classes and inheritance